[DM-4974] [L2S-1504] Release notes: strip control characters during J… (#4436)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2025.0.18-strip-control-characters-java-sdk.md

---
date: ''

> title: Add configuration option to strip control characters during JSON deserialization in Java SDK

Reviewer: ```suggestion
title: Added configuration option to strip control characters during JSON deserialization in Java SDK
```

product_area: Platform services
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: QWPx3rFfn
    label: Java SDK
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: DM-4974
version: 2025.0.18
---

> A new configuration option has been in the Java SDK that allows stripping control characters from JSON strings when deserializing responses from the API.

Reviewer: ```suggestion
A new configuration option has been introduced in the Java SDK that allows stripping control characters from JSON strings when deserializing responses from the API.
```

> Introduced a new configuration option in the Java SDK that allows stripping control characters from JSON strings when deserializing responses from the API.

Reviewer: ```suggestion
A new configuration option has been in the Java SDK that allows stripping control characters from JSON strings when deserializing responses from the API.
```

This option can be enabled by adding the property `c8y.svenson.stripControlCharacters=true` to the configuration file.

> The default value is false to maintain backward compatibility.

Reviewer 2: I would write how the application behaves in default situation - we have a Svenson library internal exception on reaching control character.

---
date: ''
title: Added configuration option to strip control characters during JSON deserialization in Java SDK
product_area: Platform services
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: QWPx3rFfn
    label: Java SDK
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: DM-4974
version: 2025.0.18
---

A new configuration option has been introduced in the Java SDK that allows stripping control characters from JSON strings when deserializing responses from the API.
This option can be enabled by adding the property `c8y.svenson.stripControlCharacters=true` to the configuration file.

> The default value is false to maintain backward compatibility.

Reviewer: ```suggestion
The default value is `false` to maintain backward compatibility.
```
